scripts/visualize/visualize.raster_map_precip.R

```diff
@@ -1,15 +1,48 @@
 visualize.raster_map_precip <- function(viz = as.viz('yahara_precip_clip')){
   
   library(ggplot2) # need to actually load this or rasterVis::gplot throws an error
+  library(sp) # need to use `as(. , "SpatialPixelsDataFrame")` so the whole library loads
   
   deps <- readDepends(viz)
-  precip_data <- deps[['raster_data']]
+  precip_raster <- deps[['raster_data']]
+  crs <- deps[["crs"]][["crs_str"]]
+  geom_feature <- deps[['geom_feature']]
+  west <- deps[["bbox"]][["west"]]
+  east <- deps[["bbox"]][["east"]]
+  north <- deps[["bbox"]][["north"]]
+  south <- deps[["bbox"]][["south"]]
+  
+  min_precip <- floor(precip_raster@data@min)
+  max_precip <- ceiling(precip_raster@data@max)
   
   png(viz[['location']])
-  map_plot <- rasterVis::gplot(precip_data, maxpixels = 5e5) + 
-    ggplot2::geom_tile(ggplot2::aes(fill = value, alpha=ifelse(is.na(value), 0, 1))) +
-    ggplot2::scale_fill_gradientn(colours=rev(rainbow(15)), na.value = "transparent",
-                         limits = c(-15, 50), 
+  
+  # get raster into correct projection
+  raster::extent(precip_raster) <- c(west, east, south, north)
+  raster::projection(precip_raster) <- sp::CRS(crs)
+  
+  # convert raster to sp object then re-project
+  precip_sp <- as(precip_raster, "SpatialPixelsDataFrame")
+  
+  # precip_sp_projected <- sp::spTransform(precip_sp, plot_crs)
+  #   bumped up against a warning about warping, looked at a stackoverflow comment
+  #   and decided to just not reproject into the plotting crs
+  #   warning: Grid warping not available, coercing to points
+  #   stackoverflow: https://stackoverflow.com/questions/15258582/re-project-sgdf-with-5-km5km-resolution-to-0-050-05
+  
+  if(!is.null(geom_feature)){
+    #crop the precip data to the watershed polygon
+    precip_sp <- raster::crop(precip_sp, geom_feature)
+  }
+  
+  precip_sp_df <- as.data.frame(precip_sp) # prep sp data
+  names(precip_sp_df) <- c("value", "x", "y")
+  
+  map_plot <- ggplot() + 
+    ggplot2::geom_tile(data = precip_sp_df,
+                       ggplot2::aes(x=x, y=y, fill = value, alpha=ifelse(is.na(value), 0, 1))) +
```

Comment:
If there is something up with the alpha values, could this be it?  Can you still put in function calls with the non-standard eval here?